File: project/seryeonha-onboarding/src/main/java/com/icd/onboarding/survey/domain/Question.java
package com.icd.onboarding.survey.domain;

import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
@Builder
@AllArgsConstructor
public class Question {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "survey_id")
    private Long surveyId;

    private Integer surveyVersion;

    @ManyToOne
    @JoinColumn(name = "survey_id", insertable = false, updatable = false)
    private Survey survey;

    private Integer orderNum;

    private String name;

    private String description;

    @Enumerated(EnumType.STRING)
    private QuestionType type;

    private boolean required;

    private boolean active;

    @OneToMany(mappedBy = "question")
    private List<Option> options = new ArrayList<>();

    @Transient
    private Integer tempId;
}

File: project/jihochoi-onboarding/QueryBox/src/main/kotlin/com/wlghsp/querybox/domain/survey/Question.kt
package com.wlghsp.querybox.domain.survey

import com.wlghsp.querybox.ui.dto.QuestionUpdateRequest
import jakarta.persistence.*
import support.domain.BaseEntity

@Entity
class Question(
    @Column(nullable = false)
    var name: String, // 항목 이름

    var description: String, // 항목 설명

    @Enumerated(EnumType.STRING)
    var type: QuestionType, // 항목 입력 형태

    required: Boolean = false, // 항목 필수 여부

    @Embedded
    var options: Options? = Options(),

    ) : BaseEntity() {

    var required: Boolean = required
        protected set

    init {
        validate()
    }

    fun validate() {
        require(name.isNotBlank()) { "항목 이름은 비어 있을 수 없습니다." }
        type.validateQuestion(options)
    }

    fun updateFrom(request: QuestionUpdateRequest) {
        validate()

        this.name = request.name
        this.description = request.description
        this.type = request.type
        this.required = request.required
    }

    companion object {
        fun from(request: QuestionUpdateRequest): Question {
            return Question(
                name = request.name,
                description = request.description,
                type = request.type,
                required = request.required,
                options = request.options?.let { Options(it.map(::Option)) }
            )
        }
    }
}
